compress.cc

```diff
@@ -90,6 +93,46 @@ shared_ptr<compressor> compressor::create(const std::map<sstring, sstring>& opti
     return {};
 }
 
+static constexpr auto uncompressed_size_marker_size = 4u;
```

Comment:
We should `static_assert` that `uncompressed_size_marker_size` is indeed the `sizeof` of the type returned by `compress_max_size`.